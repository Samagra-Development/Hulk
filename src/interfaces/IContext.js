"use strict";
//# sourceMappingURL=IContext.js.map