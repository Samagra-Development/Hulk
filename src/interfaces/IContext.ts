interface IContext {
  update(key: string, value: any);
  get(key: string);
  clear(key: string);
}