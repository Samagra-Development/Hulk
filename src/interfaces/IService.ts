interface IService {
  resolver: IServiceResolver;
  resolve(context: IContext);
  validate();
}