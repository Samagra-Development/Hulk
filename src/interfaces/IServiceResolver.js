"use strict";
//# sourceMappingURL=IServiceResolver.js.map