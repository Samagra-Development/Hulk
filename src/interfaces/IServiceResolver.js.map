{"version":3,"file":"IServiceResolver.js","sourceRoot":"","sources":["IServiceResolver.ts"],"names":[],"mappings":""}