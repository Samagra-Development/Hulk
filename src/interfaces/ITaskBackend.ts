interface ITaskBackend {
    updateTask(taskId: string, status: TaskStatus, progress: number);
    getTask(taskId: string);
    deleteTask(taskId: string);
}

export const enum TaskStatus {
    queued,
    processing
}


export type {
    ITaskBackend
}