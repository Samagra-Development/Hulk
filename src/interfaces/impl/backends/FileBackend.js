"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const fs = require("fs");
let backendFilePath;
function save(json) {
    fs.writeFileSync(backendFilePath, JSON.stringify(json));
}
class FileBackend {
    constructor(filePath) {
        backendFilePath = filePath;
        this.json = JSON.parse(fs.readFileSync(backendFilePath, 'utf-8'));
    }
    updateTask(taskId, status, progress) {
        this.json[taskId] = { status, progress };
        save(this.json);
    }
    getTask(taskId) {
        return this.json[taskId];
    }
    deleteTask(taskId) {
        delete this.json[taskId];
        save(this.json);
    }
}
exports.default = FileBackend;
//# sourceMappingURL=FileBackend.js.map