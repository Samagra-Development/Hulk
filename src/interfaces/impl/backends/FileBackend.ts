import * as fs from "fs";
import type { ITaskBackend } from "../../ITaskBackend";
import { TaskStatus } from "../../ITaskBackend";

let backendFilePath;
function save(json) {
    fs.writeFileSync(backendFilePath, JSON.stringify(json));
}

export default class FileBackend implements ITaskBackend {
    json;
    constructor(filePath: string) {
        backendFilePath = filePath;
        this.json = JSON.parse(fs.readFileSync(backendFilePath, 'utf-8'));
    }

    updateTask(taskId: string, status: TaskStatus, progress: number) {
        this.json[taskId] = { status, progress };
        save(this.json);
    }
    getTask(taskId: string) {
        return this.json[taskId];
    }
    deleteTask(taskId: string) {
        delete this.json[taskId];
        save(this.json);
    }
    
}