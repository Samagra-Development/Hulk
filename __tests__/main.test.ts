import FileBackend from "../src/interfaces/impl/backends/FileBackend";
import { TaskStatus } from "../src/interfaces/ITaskBackend";

describe('filebackend', () => {
  test("should update status", () => {
    const fileBackend = new FileBackend(__dirname + "/resources/backend.json");
    const task1 = fileBackend.getTask("1");
    // expect(task1.status).toEqual("queued");
    expect(task1).toMatchSnapshot();
    fileBackend.updateTask("2", TaskStatus.processing, 10);
    
    const task2 = fileBackend.getTask("2");
    expect(task2.progress).toEqual(10);
    expect(task2).toMatchSnapshot();
  })
});
