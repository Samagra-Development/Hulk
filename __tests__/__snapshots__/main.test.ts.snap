// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`filebackend should update status 1`] = `
Object {
  "progress": 0,
  "status": "queued",
}
`;

exports[`filebackend should update status 2`] = `
Object {
  "progress": 10,
  "status": 1,
}
`;
